safe_cli/operators/hw_accounts/ledger_manager.py

```diff
@@ -36,106 +30,30 @@ def connect(self) -> bool:
             return False
 
     @property
-    @raise_as_hw_account_exception
+    @raise_ledger_exception_as_hw_account_exception
     def connected(self) -> bool:
         """
         :return: True if ledger is connected or False in other case
         """
         return self.connect()
 
-    @raise_as_hw_account_exception
-    def get_accounts(
-        self, legacy_account: Optional[bool] = False, number_accounts: Optional[int] = 5
-    ) -> List[Tuple[ChecksumAddress, str]]:
-        """
-        Request to ledger device the first n accounts
-
-        :param legacy_account:
-        :param number_accounts: number of accounts requested to ledger
-        :return: a list of tuples with address and derivation path
-        """
-        accounts = []
-        for i in range(number_accounts):
-            if legacy_account:
-                path_string = f"44'/60'/0'/{i}"
-            else:
-                path_string = f"44'/60'/{i}'/0/0"
-
-            account = get_account_by_path(path_string, self.dongle)
-            accounts.append((account.address, account.path))
-        return accounts
-
-    @raise_as_hw_account_exception
-    def add_account(self, derivation_path: str) -> ChecksumAddress:
+    @raise_ledger_exception_as_hw_account_exception
+    def get_address_by_derivation_path(derivation_path: str) -> ChecksumAddress:
         """
-        Add an account to ledger manager set and return the added address
 
         :param derivation_path:
-        :return:
+        :return: public address for provided derivation_path
         """
-        # we should accept m/ or m'/ starting derivation paths
         if derivation_path[0:2] == "m/":
             derivation_path = derivation_path.replace("m/", "")
+        if LedgerManager.is_valid_derivation_path(derivation_path):
```

Comment:
Why are you using a static method inside the same class instead of calling `self`?